hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterMergeTask.java

```diff
@@ -78,8 +82,29 @@ public void run() {
                 executeMergeTasks(nonCoreTasks);
             }
         } finally {
-            lifecycleService.fireLifecycleEvent(joined ? MERGED : MERGE_FAILED);
+            try {
+                resetNonCoreServices();
+
+                if (joined) {
+                    tryToPromoteLocalLiteMember();
+                }
+            } finally {
+                lifecycleService.fireLifecycleEvent(joined ? MERGED : MERGE_FAILED);
+            }
+        }
+    }
+
+    private void tryToPromoteLocalLiteMember() {
+        if (node.clusterService.wasLiteMember()) {
```

Comment:
I think it's better to keep this information (`wasLiteMember` field) in `ClusterMergeTask `, instead of in `ClusterServiceImpl`.

Reply:
done